analyzers/tests/SonarAnalyzer.UnitTest/TestCases/VariableShadowsField.CSharp12.cs

```diff
@@ -0,0 +1,14 @@
+﻿class MyClass(int first, int second, int third)
+{
+    private int first; // Compliant
+
+    void MyMethod()
+    {
+        int second = 1; // Compliant
+
+        if (first is object third) // Compliant
```

Comment:
It is unclear to me why the `third` local variable defined in the pattern matching is not shadowing the `third` parameter of the primary constructor. Isn't that a false negative?

Take the following example:
```cs
class MyClass(int first, int second, int third)
{
    private int first;              // Compliant

    void MyMethod()
    {
        int first = 1;             // Noncompliant, shadows the field "first"
        int third = 1;             // Compliant?
    }
}
```
I have the impression that the shadow is not detected because primary constructor parameters are not directly visible in methods, and instead their access is replaced with access to unspeakable fields, that have different names, in the form `<paramName>P`.

So the generated code doesn't have variable shadowing. Yet it's very confusing to have a local variable named as a parameter of the primary constructor, because you may want to bring such parameter in scope, and you can't if the parameter is shadowed.

Reply:
As discussed offline, we think the rule should be extended to consider primary constructors, thus all of the UTs above are FN.